app/controllers/carts_controller.rb

```diff
@@ -1,20 +1,31 @@
 class CartsController < ApplicationController
 
+  before_action :find_cart
+
   def add
     dish = Dish.find(params[:id])
-    quantity = params[:amount].nil? ? 1 : params[:amount]
-    if session[:cart_id]
-      show
+    if /(\D+)/.match(params[:amount]).nil?
+      @cart.add(dish, dish.price, params[:amount].to_i)
+      flash[:notice] = "#{dish.name} added to cart: #{params[:amount]}"
+    else
+      flash[:notice] = "Please enter a number"
+    end
+    redirect_back(fallback_location: (request.referer || restaurant_path(params[:restaurant_id])))
+  end
+
+
+  def remove_item
+    dish = Dish.find(params[:dish_id])
+    if /(\D+)/.match(params[:remove_amount]).nil?
```

Comment:
Same here. If you need more complicated check, extract it to a separate method to avoid code duplication